internal/plugin/dataplane_server.go

```diff
@@ -29,32 +29,19 @@ type (
 		Message string `json:"message,omitempty"`
 	}
 
-	DataplaneServerParameters struct {
-		Host            string
-		Port            int
-		Logger          *slog.Logger
-		instanceService service.InstanceServiceInterface
-	}
-
 	DataplaneServer struct {
-		address         string
-		logger          *slog.Logger
-		instanceService service.InstanceServiceInterface
-		instances       []*instances.Instance
-		messagePipe     bus.MessagePipeInterface
-		server          net.Listener
+		address     string
+		logger      *slog.Logger
+		instances   []*instances.Instance
+		messagePipe bus.MessagePipeInterface
+		server      net.Listener
 	}
 )
 
-func NewDataplaneServer(dataplaneServerParameters *DataplaneServerParameters) *DataplaneServer {
-	if dataplaneServerParameters.instanceService == nil {
-		dataplaneServerParameters.instanceService = service.NewInstanceService()
-	}
-
+func NewDataplaneServer(agentConfig *config.Config, logger *slog.Logger) *DataplaneServer {
 	return &DataplaneServer{
-		address:         fmt.Sprintf("%s:%d", dataplaneServerParameters.Host, dataplaneServerParameters.Port),
-		logger:          dataplaneServerParameters.Logger,
-		instanceService: dataplaneServerParameters.instanceService,
+		address: fmt.Sprintf("%s:%d", agentConfig.DataplaneAPI.Host, agentConfig.DataplaneAPI.Port),
```

Comment:
Should we use `net.JoinHostPort()` as it can deal with IPv6 hosts? https://pkg.go.dev/net#JoinHostPort
```suggestion
		address: net.JoinHostPort(agentConfig.DataplaneAPI.Host, strconv.FormatInt(agentConfig.DataplaneAPI.Port), 10),
```